mesonbuild/mtest.py

```diff
@@ -228,11 +228,15 @@ def run_single_test(self, wrap, test):
 
             # If MALLOC_PERTURB_ is not set, or if it is set to an empty value,
             # (i.e., the test or the environment don't explicitly set it), set
-            # it ourselves. We do this unconditionally because it is extremely
-            # useful to have in tests.
+            # it ourselves. We do this unconditionally for regular tests
+            # because it is extremely useful to have. For benchmarks this is
+            # set to zero as it would decrease runtime performance.
             # Setting MALLOC_PERTURB_="0" will completely disable this feature.
             if 'MALLOC_PERTURB_' not in child_env or not child_env['MALLOC_PERTURB_']:
-                child_env['MALLOC_PERTURB_'] = str(random.randint(1, 255))
+                if self.options.benchmark:
+                    child_env['MALLOC_PERTURB_'] = '0'
```

Comment:
Would it better to not touch `MALLOC_PERTURB_` at all if benchmark is true?

Reply:
Well, if `MALLOC_PERTURB_` is set, it is not changed anyway. But we could also simply not touch it. I can update the PR if you want.

Reply:
The code is simpler to understand if you only change it if not benchmark. Please do so.